oap-server/server-receiver-plugin/skywalking-log-recevier-plugin/src/main/java/org/apache/skywalking/oap/server/recevier/log/provider/handler/LogReportServiceHandler.java

```diff
@@ -0,0 +1,93 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.skywalking.oap.server.recevier.log.provider.handler;
+
+import io.grpc.stub.StreamObserver;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.skywalking.apm.network.common.v3.Commands;
+import org.apache.skywalking.apm.network.logging.v3.LogData;
+import org.apache.skywalking.apm.network.logging.v3.LogReportServiceGrpc;
+import org.apache.skywalking.oap.log.analyzer.module.LogAnalyzerModule;
+import org.apache.skywalking.oap.log.analyzer.provider.log.ILogAnalyzerService;
+import org.apache.skywalking.oap.server.library.module.ModuleManager;
+import org.apache.skywalking.oap.server.library.server.grpc.GRPCHandler;
+import org.apache.skywalking.oap.server.telemetry.TelemetryModule;
+import org.apache.skywalking.oap.server.telemetry.api.CounterMetrics;
+import org.apache.skywalking.oap.server.telemetry.api.HistogramMetrics;
+import org.apache.skywalking.oap.server.telemetry.api.MetricsCreator;
+import org.apache.skywalking.oap.server.telemetry.api.MetricsTag;
+
+@Slf4j
+public class LogReportServiceHandler extends LogReportServiceGrpc.LogReportServiceImplBase implements GRPCHandler {
+
+    private final ModuleManager moduleManager;
+    private HistogramMetrics histogram;
+    private CounterMetrics errorCounter;
+    private ILogAnalyzerService logAnalyzerService;
+
+    public LogReportServiceHandler(final ModuleManager moduleManager) {
+        this.moduleManager = moduleManager;
+        MetricsCreator metricsCreator = moduleManager.find(TelemetryModule.NAME)
+                                                     .provider()
+                                                     .getService(MetricsCreator.class);
+        this.logAnalyzerService = moduleManager.find(LogAnalyzerModule.NAME)
+                                               .provider()
+                                               .getService(ILogAnalyzerService.class);
+
+        histogram = metricsCreator.createHistogramMetric(
+            "log_in_latency", "The process latency of log",
+            new MetricsTag.Keys("protocol"), new MetricsTag.Values("grpc")
+        );
+        errorCounter = metricsCreator.createCounter("log_analysis_error_count", "The error number of log analysis",
+                                                    new MetricsTag.Keys("protocol"), new MetricsTag.Values("grpc")
+        );
+    }
+
+    @Override
+    public StreamObserver<LogData> collect(final StreamObserver<Commands> responseObserver) {
+        return new StreamObserver<LogData>() {
+            @Override
+            public void onNext(final LogData logData) {
+                if (log.isDebugEnabled()) {
+                    log.debug("received log in streaming");
+                }
+                HistogramMetrics.Timer timer = histogram.createTimer();
+                try {
+                    logAnalyzerService.doAnalysis(logData);
```

Comment:
Please note this at the protocol level, https://github.com/apache/skywalking-data-collect-protocol/blob/345d9edac842179e61504999c91bf2bec6ae0ff7/logging/Logging.proto#L50

Also, this only works as the streaming mode in the gRPC, or batch mode of HTTP receivers. It isn't suitable for the queue.

Reply:
done